Assets/Scripts/Teleport.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Teleport : MonoBehaviour
+{
+    public GameObject player;
+    public GameObject teleporter;
+    public Transform destination;
+    public int colorValue = 0;
+    public double timer;
+    public bool stop;
+
+    // Use this for initialization
+    void Start ()
+    {
+        destination = teleporter.transform;
+        GetComponent<Renderer>().material.color = gameObject.GetComponent<Renderer>().material.GetColor("_Color");
+        timer = 1000000000001;
+    }
+
+    // Update is called once per frame
+    void Update ()
+    {
+        //Debug.Log(ColorSwap.players[0].currentPlayerColor);
+
+        //The purpose of the timer is so that it doesn't teleport the player infinitely.
+        if (timer > 0 && timer < 1000000000000)
+        {
+            timer += Time.deltaTime;
+        }
```

Comment:
It'd be a bit cleaner to have the timer count down rather than up. This way you can initialize it to 0, set it to 0.3 on teleport, and have it count down to 0 again